pennylane/transforms/core/transform_program.py

```diff
@@ -109,3 +197,58 @@ def is_empty(self):
             bool: Boolean, True if empty, False otherwise.
         """
         return len(self) == 0
+
+    @property
+    def is_informative(self) -> bool:
+        """Check if the transform program is informative or not.
```

Comment:
kinda unrelated to this PR, but what's the point of informative transforms?

Reply:
The idea is that we can use the same infrastructure for things like `draw`, `specs`, `commutation_dag`, etc.

I'm just now a little unsure about that idea, since postprocessing functions assume everything is a result object. I don't want to start mixing up things that are not result objects with things that are.  Or having a function like `QNode.__call__` that's supposed to return results, and returning a different type of thing.